tests/test_data_preparation.py

```diff
@@ -24,25 +34,50 @@ def compute_md5(file_path):
     return hash_md5.hexdigest()
 
 
+def docker_rm_and_mkdir(file_):
+    directory = Path(file_).absolute().parent
+    test_config_path = Path(__file__).absolute().parent / "gpu-tests" / "test-local.yaml"
+    config = yaml.safe_load(open(test_config_path).read())
+    volumes = config['mounts']
+    container = config['containers']['nemo-skills']
+    rm_cmd = f"rm -f {str(file_)}"
+    mkdir_cmd = f"mkdir -p {str(directory)}"
+    docker_run(
+        image_name=container,
+        volume_paths=volumes,
+        command=rm_cmd,
+    )
+    docker_run(
+        image_name=container,
+        volume_paths=volumes,
+        command=mkdir_cmd,
+    )
+
+
 def test_multiple_files():
-    output_file = "tests/data/processed_multifile_output.jsonl"
-    subprocess.run(
-        "python -m nemo_skills.training.prepare_sft_data "
-        f"    ++input_files='tests/data/output-rs*.test' "
-        f"    ++output_path={output_file} "
-        f"    ++prompt_config=generic/math "
-        f"    ++prompt_template=llama3-instruct "
-        f"    ++exclude_optional_keys=false "
-        f"    ++filters.remove_len_outlier_problems=false "
-        f"    ++filters.drop_multi_boxed=true "
-        f"    ++filters.trim_solutions=true "
-        f"    ++filters.drop_incorrect_arithmetic=false "
-        f"    ++filters.split_arithmetic=false "
-        f"    ++num_output_samples=32 "
-        f"    ++downsampling_method=fair "
-        f"    ++do_shuffle=false ",
-        check=True,
-        shell=True,
+    output_file = f"/tmp/nemo-skills-tests/data/processed_multifile_output.jsonl"
+    docker_rm_and_mkdir(output_file)
+    run_cmd(
+        module="nemo_skills.training.prepare_sft_data ",
+        cluster='test-local',
+        config_dir=Path(__file__).parent / 'gpu-tests',
+        log_dir='/tmp/nemo-skills-tests/test_multiple_files',
+        num_gpus=0,
```

Comment:
Does it not work with default value of None? If so, we need to fix it, it shouldn't be necessary to set it to 0 explicitly

Reply:
Slurm errors if you try to queue a job on the GPU partition without GPUs set, with an error like:

```
sbatch: error: Cannot find GPU specification, you may not submit a job not requesting GPUs in a non-CPU partition, partition: ***
sbatch: error: Batch job submission failed: Unspecified error
```

Therefore, I set the default value of gpus to `1` for `run_cmd`, to avoid this issue--if for some reason you didn't have GPUs, you could override to 0.

Alternatively, if you want the default to be None, that is fine, we'll just always have to set it when running on slurm.

Reply:
Can we set it to None by default in the arguments of run_cmd and then check before add_task and set to 0 if cluster_config['executor'] == local and 1 otherwise? That way the default will work for both slurm and local and people can explicitly set any value they want to override

Reply:
actually, let's do it inside add_task itself as it's not specific to run_cmd, I guess

Reply:
fixed in 126e43d2f87616d0ecda2bdd1fda544ebf9af314

note: don't need to set to 0 for the docker executor since it falls back to 0 when num_gpus is None